File: src/com/will/arraysAndHashing/ContainsDuplicate.java
package com.will.arraysAndHashing;

import java.util.*;

//https://leetcode.com/problems/contains-duplicate/

/**
 * Given an integer array nums, return true if any value appears at least twice in the array, and return false if every element is distinct.
 */
class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hm = new HashSet<>();
        for (int i: nums) {
            if(hm.contains(i)){
                return true;
            }
            hm.add(i);
        }
        return false;
    }
}

File: python/solutions/contains_duplicate.py


class Solution:
    def containsDuplicate(self, nums): 
        mySet = set()
        for n in nums:
            if n in mySet:
                return True
            else:
                mySet.add(n)
                
        return False
    

nums = [1,2,3,1]

solution = Solution()

print(solution.containsDuplicate(nums))